app/app.go

```diff
@@ -297,66 +283,37 @@ func NewAxelarApp(
 
-	// Create static IBC router, add axelarnet module as the IBC transfer route, and seal it
-	ibcRouter := porttypes.NewRouter()
-	ibcRouter.AddRoute(ibctransfertypes.ModuleName, axelarnetModule)
+	wasmDir := filepath.Join(homePath, "wasm")
+	wasmConfig := mustReadWasmConfig(appOpts)
 
-	// axelar custom keepers
-	// axelarnet / evm / nexus keepers created above
-	setKeeper(keepers, initRewardKeeper(appCodec, keys, keepers))
-	setKeeper(keepers, initMultisigKeeper(appCodec, keys, keepers))
-	setKeeper(keepers, initTssKeeper(appCodec, keys, keepers))
-	setKeeper(keepers, initSnapshotKeeper(appCodec, keys, keepers))
-	setKeeper(keepers, initVoteKeeper(appCodec, keys, keepers))
-	setKeeper(keepers, initPermissionKeeper(appCodec, keys, keepers))
+	scopedWasmK := capabilityK.ScopeToModule(wasm.ModuleName)
+	wasmK := initWasmKeeper(appCodec, keys, keepers, scopedWasmK, bApp, wasmDir, wasmConfig, wasmOpts)
+	setKeeper(keepers, wasmK)
 
-	var wasmK wasm.Keeper
+	setKeeper(keepers, initGovernanceKeeper(appCodec, keys, keepers))
 
-	wasmDir := filepath.Join(homePath, "wasm")
-	wasmConfig := mustReadWasmConfig(appOpts)
+	capabilityK.Seal()
 
-	if IsWasmEnabled() {
+	// set the contract keeper for the Ics20WasmHooks
+	wasmHooks.ContractKeeper = wasmkeeper.NewDefaultPermissionKeeper(wasmK)
 
-		// The last arguments can contain custom message handlers, and custom query handlers,
-		// if we want to allow any custom callbacks
-		wasmOpts = append(wasmOpts, wasmkeeper.WithMessageHandlerDecorator(func(old wasmkeeper.Messenger) wasmkeeper.Messenger {
-			return wasmkeeper.NewMessageHandlerChain(old, nexusKeeper.NewMessenger(getKeeper[nexusKeeper.Keeper](keepers)))
-		}))
-		wasmK = wasm.NewKeeper(
-			appCodec,
-			keys[wasm.StoreKey],
-			keepers.getSubspace(wasm.ModuleName),
-			getKeeper[authkeeper.AccountKeeper](keepers),
-			getKeeper[bankkeeper.BaseKeeper](keepers),
-			getKeeper[stakingkeeper.Keeper](keepers),
-			getKeeper[distrkeeper.Keeper](keepers),
-			getKeeper[*ibckeeper.Keeper](keepers).ChannelKeeper,
-			getKeeper[*ibckeeper.Keeper](keepers).ChannelKeeper,
-			&getKeeper[*ibckeeper.Keeper](keepers).PortKeeper,
-			scopedWasmK,
-			getKeeper[ibctransferkeeper.Keeper](keepers),
-			bApp.MsgServiceRouter(),
-			bApp.GRPCQueryRouter(),
-			wasmDir,
-			wasmConfig,
-			WasmCapabilities,
-			wasmOpts...,
-		)
+	messageRouter := nexusTypes.NewMessageRouter().
+		AddRoute(evmTypes.ModuleName, evmKeeper.NewMessageRoute()).
+		AddRoute(axelarnetTypes.ModuleName, axelarnetKeeper.NewMessageRoute(getKeeper[axelarnetKeeper.Keeper](keepers), getKeeper[axelarnetKeeper.IBCKeeper](keepers), getKeeper[feegrantkeeper.Keeper](keepers), axelarbankkeeper.NewBankKeeper(getKeeper[bankkeeper.BaseKeeper](keepers)), getKeeper[nexusKeeper.Keeper](keepers), getKeeper[authkeeper.AccountKeeper](keepers)))
```

Comment:
this is a bit messy (verbose), is there any way it can be cleaned up a bit?

Reply:
I can put the arguments on separate lines, otherwise there is not much I can do